src/switch_core_io.c

```diff
@@ -194,10 +194,23 @@ SWITCH_DECLARE(switch_status_t) switch_core_session_read_frame(switch_core_sessi
 		}
 
 		if (status == SWITCH_STATUS_INUSE) {
+			switch_bool_t continue_recording = switch_channel_var_true(session->channel, "continue_recording_on_hold");
```

Comment:
`continue_recording_on_hold` isn't really an indication that there is a recording bug later on or probably it was already been stop. I think it would be best is to check if there is a flag in the media bug that requires this instead of a channel var and only send the dummy frame into that media bug that requires this. The issue here is that if this is set then it can also be apply to other media bug and not just recording. For example, telnyx_amd or telnyx_async_dtmf use the frame to calculate time so if the user hold the media then the timer from those bug will continue to run.